source/code-snippets/collation/findOneAndDelete-collation.js

```diff
@@ -0,0 +1,58 @@
+// ignored first line
+// start findOneAndDelete example without collation
+const { MongoClient } = require("mongodb");
+
+// Replace the uri string with your MongoDB deployment's connection string.
+const uri =
+  "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority&useUnifiedTopology=true";;
+
+// Create a new MongoClient
+const client = new MongoClient(uri);
+
+async function run() {
+  try {
+    // Connect the client to the server
+    await client.connect();
+
+    const db = client.db("myproject");
+    const collection = db.collection("numbers");
+
+    const result = await collection.findOneAndDelete({ a: { $gt: "100" } });
+
+    console.dir(result);
+  } finally {
+    // Ensures that the client will close when you finish/error
+    await client.close();
+  }
+}
+run().catch(console.dir);
+// end findOneAndDelete example without collation
+
+// start findOneAndDelete example with collation
+const uri =
+  "mongodb+srv://sample-hostname:27017/?poolSize=20&useUnifiedTopology=true";
+
+// Create a new MongoClient
+const client = new MongoClient(uri);
+
+async function run() {
+  try {
+    // Connect the client to the server
+    await client.connect();
+
+    const db = client.db("myproject");
+    const collection = db.collection("numbers");
+
+    const result = await collection.findOneAndDelete(
+      { a: { $gt: "100" } },
+      { collation: { locale: "en", numericOrdering: true } },
```

Comment:
If `locale` isn't required here, I'd suggest removing it since it distracts uses from `numericOrdering` which is the point of this example. If it is required, can we at least switch the order of the collation fields so `numericOrdering` comes first?

Reply:
Good recommendation to shorten the example, but per the documentation, the `locale` field is always required. 😛

Reply:
Regardless, the order is a great point. I'm not sure there's an advantage with playing with the order. I've always placed required params first in associative arrays/maps followed by optional ones which is similar to parameter ordering conventions in several programming languages.

Reply:
Yeah I can definitely understand listing it first -- it's also a parameter that users are already familiar with, so not a problem at all IMO. Leaving it as is seems perfectly fine!